samples/bluetooth/mesh/sensor_server/src/model_handler.c

```diff
@@ -266,26 +357,32 @@ static void button_handler_cb(uint32_t pressed, uint32_t changed)
 		return;
 	}
 
-	if (pressed & BIT(0)) {
-		int err;
+	if (pressed & changed & BIT(0)) {
+		k_work_reschedule(&presence_detected_work, K_MSEC(100 * pres_mot_thres.val1));
+	}
 
-		/* This sensor value must be boolean -
-		 * .val1 can only be '0' or '1'
-		 */
-		struct sensor_value val = {
-			.val1 = 1,
-		};
+	if ((!pressed) & changed & BIT(0)) {
+		if (prev_pres) {
```

Comment:
I don't think this will work to scale the button pressing time by motion threshold. I think you will have to sample uptime when k_work is scheduled and then compare it with current and check how many seconds have elapsed.

Reply:
I think, the logic to compute time since presence detected is still incorrect. Let's imagine following sequence of actions:
1. Send Sensor Get to fetch 'Time since presence detected'. Received response will be 0 (or 0xFFFF).
2. Wait for 60 seconds or more.
2. Press and hold button to detect the presence. 
3. When detection delayed work triggers (i.e. successful presence detection), the "[prev_detect = 0](https://github.com/nrfconnect/sdk-nrf/pull/10943/files#diff-7fd712d674be1168023ebf7ca743f337b1cf32f4aca217e754b1abd972175c96R351)" here will cause old timestamp to be erased. 
4. Again, quickly send Sensor Get to fetch 'Time since presence detected'. Now, you should something less than 60 seconds, however, the "[prev_detect = 0](https://github.com/nrfconnect/sdk-nrf/pull/10943/files#diff-7fd712d674be1168023ebf7ca743f337b1cf32f4aca217e754b1abd972175c96R351)" will cause reported time to be from the beginning of the system boot (>60 seconds), which will be obviously incorrect.

Reply:
This will work like this:
1. Send Sensor Get to fetch "Time since presence detected". Received response will be x seconds since last detection.
2. Press and hold button to detected the presence. Detection presence.
3. Wait for 60 seconds.
4. Send Sensor Get to fetch "Time since presence detected". Received response will be 0 seconds since last detection as presence is active.
5. Wait for 60 seconds.
6. Release button to represent no presence presence. No presence.
7. Send Sensor Get to fetch "Time since presence detected". Received response will be x seconds since last detection.

The question is last present running from when presence is started detected, or when presence is no longer active. I would assume when the "Time since present detected" is from when presence is no longer present. If not then the time for something moving will be added to the time, and that is a little bit strange to add.
I may give the effect that if someplace has a movement over time and "Time since presence detected" is ex. 5 min it will turn of the light, instead of after 5 min without movement.

Reply:
It will be 0 as long as presence is active, last presence is 0 as long as it is in presence detected. Time should be set when detection is no longer detected. See scenario bellow.